ap/calibration/seeds.py

```diff
@@ -0,0 +1,112 @@
+# coding: utf-8
+
+"""
+Methods related to creating event and object seeds during calibration.
+"""
+
+import hashlib
+
+from ap.calibration import calibrator
+from ap.util import maybe_import, primes
+from ap.columnar_util import Route, get_ak_indices, set_ak_column, has_ak_column
+
+np = maybe_import("numpy")
+ak = maybe_import("awkward")
+
+
+@calibrator(
+    uses={
+        # global columns for event seed
+        "event", "nGenJet", "nGenPart", "nJet", "nPhoton", "nMuon", "nElectron", "nTau", "nSV",
+        # first-object columns for event seed
+        "Tau.jetIdx", "Tau.decayMode",
+        "Muon.jetIdx", "Muon.nStations",
+        "Jet.nConstituents", "Jet.nElectrons", "Jet.nMuons",
+    },
+    produces={"deterministic_seed"},
+)
+def deterministic_event_seeds(events, create_seed, **kwargs):
+    # start with a seed of the event number itself as the most sensitive integer
+    seed = create_seed(events.event)
+
+    # get global integers
+    global_fields = ["nGenJet", "nGenPart", "nJet", "nPhoton", "nMuon", "nElectron", "nTau", "nSV"]
+    for i, f in enumerate(global_fields, 3):
```

Comment:
Where is this number `3` coming from? I assume it's an arbitrary offset of some sort? Perhaps it would be nice to make this a variable that is centrally initialized somewhere (or even an argurment to this function with some default value). If it's not arbitrary, could you perhaps provide a comment on where this is coming from?

Reply:
✓